src/mgr/PyModule.cc

```diff
@@ -82,6 +92,34 @@ std::string handle_pyerror()
         }
     }
     formatted = str("").join(formatted_list);
+
+    if (crash_dump) {
+      std::list<std::string> bt_strings;
+      std::map<std::string, std::string> extra;
+
+      extra["mgr_module"] = module;
```

Comment:
i'd create a `PyBackTrace` class which collects the necessary bits and backtrace in its ctor and exposes a `dump(Formatter *f)` method. we could have a templatized `generate_crash_dump()` or a parent class shared by `BackTrace` (probably should be renamed to `CxxBackTrace`) and `PyBackTrace`, so it can be used by both signal handler and python exception handling.